Update the device entry in glossary

c0ffeeca7 opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->
- Rephrase the device entry in the glossary. 
- Add an additional example.
- Add information about properties.
- Add link to related developer documentation.

I saw that the links in the other glossary entries all point to user documentation, not developer documentation. So I'm not sure if we want to include this link. My rational was that developers are also a target audience of these entries.

## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

c0ffeeca7 committed:
Update the device entry in glossary

c0ffeeca7 committed:
Add line breaks, fix link

c0ffeeca7 committed:
Remove line breaks

- as they do not seem to work
- improve example titles

frenck commented on source/_data/glossary.yml:
```diff
@@ -69,15 +69,16 @@
-    for a device would be a smart plug named 'Coffee Machine' which provides
-    a `switch` entity plus one or more `sensor` entities for power monitoring
-    or similar.
+  definition: >    
```
```suggestion
  definition: >
```

frenck commented on source/_data/glossary.yml:
```diff
@@ -69,15 +69,16 @@
-    or similar.
+  definition: >    
+    A device is a model representing a physical or logical unit that contains entities and has its own control unit.
+    <b>Example for a device as a physical unit</b>
```
```suggestion
    **Example for a device as a physical unit**
```

frenck commented on source/_data/glossary.yml:
```diff
@@ -69,15 +69,16 @@
+    A device is a model representing a physical or logical unit that contains entities and has its own control unit.
+    <b>Example for a device as a physical unit</b>
+    A smart plug named 'Coffee machine' which provides 2 entities: a `switch` entity to turn power on or off ('Coffee machine power switch') and a `sensor` entity for power monitoring ('Coffee machine power sensor').
+    <b>Example for a device as a logical unit</b>
```
```suggestion
    **Example for a device as a logical unit**
```

frenck commented on source/_data/glossary.yml:
```diff
@@ -69,15 +69,16 @@
+    <b>Example for a device as a physical unit</b>
+    A smart plug named 'Coffee machine' which provides 2 entities: a `switch` entity to turn power on or off ('Coffee machine power switch') and a `sensor` entity for power monitoring ('Coffee machine power sensor').
+    <b>Example for a device as a logical unit</b>
+    An Ecobee thermostat with 4 room sensors. This thermostat is seen as 5 devices in Home Assistant: 1 device for the thermostat with 4 sensors, and 1 device for each room sensor. Each device can be in a different area, and may have more than one input or output within that area.
```
```suggestion
    An ecobee thermostat with 4 room sensors. This thermostat is seen as 5 devices in Home Assistant: 1 device for the thermostat with 4 sensors, and 1 device for each room sensor. Each device can be in a different area and may have more than one input or output within that area.
```

frenck commented on source/_data/glossary.yml:
```diff
@@ -69,15 +69,16 @@
-    a `switch` entity plus one or more `sensor` entities for power monitoring
-    or similar.
+  definition: >    
+    A device is a model representing a physical or logical unit that contains entities and has its own control unit.
```
What is a "control unit" ?

c0ffeeca7 replied:
I don't know :-). I found this in the documentation: (_A device in Home Assistant represents either a physical device that has its own control unit_) and added it because I though it was a mandatory component of any device.

frenck commented on source/_data/glossary.yml:
```diff
@@ -69,15 +69,16 @@
+    <b>Example for a device as a logical unit</b>
+    An Ecobee thermostat with 4 room sensors. This thermostat is seen as 5 devices in Home Assistant: 1 device for the thermostat with 4 sensors, and 1 device for each room sensor. Each device can be in a different area, and may have more than one input or output within that area.
+    Devices have properties such as ID, manufacturer, name, model, hardware version, firmware version, connections, etc.
+  link: https://developers.home-assistant.io/docs/device_registry_index
```
I'm not a fan of linking user documentation to developer documentation for these things. In general, it becomes really complex really fast. If there is information on that page we are missing in our end-user docs, we should add that to our end-user documentation.

c0ffeeca7 replied:
I understand. I was hesitant to add the link here because I realize the primary target audience for the user documentation are the users, not developers.

c0ffeeca7 committed:
Update source/_data/glossary.yml

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

c0ffeeca7 committed:
Update source/_data/glossary.yml

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

c0ffeeca7 committed:
Update source/_data/glossary.yml

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

c0ffeeca7 committed:
Update source/_data/glossary.yml

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

c0ffeeca7 committed:
Remove link to developer documentation

- since this is user documentation, the links should not point to developer documentation, as users are the primary target audience.
- remove control unit

frenck committed:
Fix display of glossaries

frenck commented on source/_data/glossary.yml:
```diff
@@ -27,60 +27,63 @@
   excerpt: >
-    A device is a named collection of entities that all represent the same
-    physical/logical unit, which can do or observe something.
+    A device is a model representing a physical or logical unit that contains entities and has its own control unit.
```
```suggestion
    A device is a model representing a physical or logical unit that contains entities.
```

frenck reviewed (changes requested):
One small change left, to match the change on the main definition.

c0ffeeca7 committed:
Remove 'has its own control unit'

Co-authored-by: Franck Nijhof <frenck@frenck.nl>
